tests/test_swift.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+
+# Any copyright is dedicated to the Public Domain.
+# http://creativecommons.org/publicdomain/zero/1.0/
+
+from pathlib import Path
+import shutil
+import subprocess
+
+from glean_parser import swift
+from glean_parser import metrics
+from glean_parser import pings
+from glean_parser import translate
+
+
+ROOT = Path(__file__).parent
+
+
+def test_parser(tmpdir):
+    """Test translating metrics to Swift files."""
+    tmpdir = Path(tmpdir)
+
+    translate.translate(
+        ROOT / "data" / "core.yaml",
+        "swift",
+        tmpdir,
+        {},
+        {"allow_reserved": True},
+    )
+
+    assert set(x.name for x in tmpdir.iterdir()) == set(
+        [
+            "CorePing.swift",
+            "Telemetry.swift",
+            "Environment.swift",
+            "DottedCategory.swift",
+            "GleanInternalMetrics.swift",
+        ]
+    )
+
+    # Make sure descriptions made it in
+    with open(tmpdir / "CorePing.swift", "r", encoding="utf-8") as fd:
+        content = fd.read()
+        assert "True if the user has set Firefox as the default browser." in content
+
+    with open(tmpdir / "Telemetry.swift", "r", encoding="utf-8") as fd:
+        content = fd.read()
+        assert "جمع 搜集" in content
+
+    with open(tmpdir / "GleanInternalMetrics.swift", "r", encoding="utf-8") as fd:
+        content = fd.read()
+        assert 'category: ""' in content
+
+    # Only run this test if ktlint is on the path
+    if shutil.which("ktlint"):
+        for filepath in tmpdir.glob("*.swift"):
+            subprocess.check_call(["ktlint", filepath])
+
+
+def test_swift_generator():
+    kdf = swift.swift_datatypes_filter
+
+    assert kdf("\n") == r'"\n"'
+    assert kdf([42, "\n"]) == r'[42, "\n"]'
+    assert (
+        kdf({"key": "value", "key2": "value2"})
+        == r'["key": "value", "key2": "value2"]'
+    )
+    assert kdf(metrics.Lifetime.ping) == ".ping"
+
+
+def test_metric_type_name():
+    event = metrics.Event(
```

Comment:
As far as I can see, the only metric type available in the iOS implementation is the `CounterMetricType`. Let's just account for it, leaving all the other types out: we can land this as an initial PR and iterate on it as needed.